### PR TITLE
Updates/new protect from styles

### PR DESCRIPTION
**Pull request by enejb**
Update the styling of the Protect page. cc: @rickybanister , @MichaelArestad, @jeffgolenski 
Adding the following constant `define( 'JETPACK_ALWAYS_PROTECT_LOGIN', true );` helps with testing.

#### Changes proposed in this Pull Request:
The new UI Looks like this (latest 24px padding screenshot at the bottom)

The new protect page when you are blocked.
screen shot 2017-12-08 at 11 04 59 am

After the email is sent to you successfully:
screen shot 2017-12-08 at 11 11 02 am

Login page ( after you click on the link)
screen shot 2017-11-29 at 11 00 55 am

The token has expired or someone is trying to guess the token.
screen shot 2017-12-08 at 9 55 49 am

Email was entered wrong.
screen shot 2017-12-08 at 10 10 48 am

Email already sent 
screen shot 2017-12-08 at 10 03 41 am

#### Testing instructions:
* The same instuction original

### CODE REVIEW

**Comment by enejb** [comment]
- [ ] Update the graphic to that it looks sharp. 
- [ ] Update the graphic in the email.

**Comment by rickybanister** [comment]
I think we're very close to merging. I noticed a couple of small things—

1. Can the errors have something similar to the `notice` styling and be red? 
2. Can the success message say `Recovery instructions were sent to %s. Check your inbox!`

Thanks!

I'm working on the svg, but that can wait til after 5.6 if need be.

**Comment by enejb** [comment]
Are you taking about a notice like this? 
screen shot 2017-11-29 at 11 35 10 am

**Comment by enejb** [comment]
Update with the new wording

File: modules/protect/blocked-login-page.php [hunk 1]
```diff
@@ -220,9 +220,12 @@ public function render_blocked_login_message() {
 
 	function process_recovery_email() {
 		$sent = $this->send_recovery_email();
-
+		$show_recovery_form = true;
 		if ( is_wp_error( $sent ) ) {
-			$this->protect_die( $sent, true );
+			if( $sent->get_error_code() !== 'email_already_sent' ) {
```
Comment (jeherve) [inline modules/protect/blocked-login-page.php]: Can we use Yoda conditions here?

File: modules/protect/blocked-login-page.php [hunk 2]
```diff
@@ -220,9 +220,12 @@ public function render_blocked_login_message() {
...
+				$show_recovery_form = true;
+			}
+			$this->protect_die( $sent,  null,true, $show_recovery_form );
```
Comment (jeherve) [inline modules/protect/blocked-login-page.php]: Something seems off with spacing here.

File: modules/protect/blocked-login-page.php [hunk 3]
```diff
@@ -192,7 +192,7 @@ public function render_and_die() {
 		}
 
 		if ( isset( $_GET['validate_jetpack_protect_recovery'] ) && $_GET['user_id'] ) {
-			$this->protect_die( __( 'Could not validate recovery token.', 'jetpack' ) );
+			$this->protect_die( __( 'Oops, we couldn’t validate the recovery token.', 'jetpack' ) );
```
Comment (jeherve) [inline modules/protect/blocked-login-page.php]: That `’` character may cause some confusion down the road. Should we just use `'`?

File: modules/protect/blocked-login-page.php [hunk 4]
```diff
@@ -236,7 +239,7 @@ function send_recovery_email() {
 		$user = get_user_by( 'email', trim( $email ) );
 
 		if ( ! $user ) {
-			return new WP_Error( 'invalid_user', __( 'Oops, could not find a user with that email address.', 'jetpack' ) );
+			return new WP_Error( 'invalid_user', __( 'Oops, we couldn’t find a user with that email. Please try again!', 'jetpack' ) );
```
Comment (jeherve) [inline modules/protect/blocked-login-page.php]: That `’` character may cause some confusion down the road. Should we just use `'`?

File: modules/protect/blocked-login-page.php [hunk 5]
```diff
@@ -258,72 +261,60 @@ function send_recovery_email() {
...
 	}
 
-	function protect_die( $content, $back_link = false ) {
-		$image = sprintf(
-			'<img src="%s" width="180" style="display: block; margin: 0 auto;" />',
-			plugins_url( 'modules/protect/jetpack-security.png', JETPACK__PLUGIN_FILE )
-		);
-
+	function protect_die( $content, $title = null, $back_link = false, $recovery_form = false ) {
+		if( empty( $title ) ) {
```
Comment (jeherve) [inline modules/protect/blocked-login-page.php]: Missing space here.

File: modules/protect/blocked-login-page.php [hunk 6]
```diff
@@ -258,72 +261,60 @@ function send_recovery_email() {
...
+		$content =  '<p>'. $content .'</p>';
 
+		// If for some reason the login pop up box show up in the wp-admin.
 		if ( isset( $_GET['interim-login'] ) ) {
-			$content = "<style>html{ background-color: #fff; } #error-page { margin:0 auto; padding: 1em; box-shadow: none; } </style>" . $content;
+			$content = "<style>html{ background-color: #fff; } #error-message { margin:0 auto; padding: 1em; box-shadow: none; } </style>" . $content;
 		}
-
-		wp_die( $content, $this->page_title, array( 'back_link' => $back_link ) );
+		$this->display_page( $title, $content, 'error message goes here', $back_link, $recovery_form );
```
Comment (jeherve) [inline modules/protect/blocked-login-page.php]: `'error message goes here'` should probably be updated.

File: modules/protect/blocked-login-page.php [hunk 7]
```diff
@@ -258,72 +261,60 @@ function send_recovery_email() {
...
 				<?php echo wp_nonce_field( 'bypass-protect' ); ?>
-                <p><label for="email" style="font-size:12px;">Email Address<br/></label>
-                    <input type="email" name="email" style="font-size:24px; padding:3px; margin: 2px 6px 16px 0; width:100%; border: 1px solid #ddd;
-    box-shadow: inset 0 1px 2px rgba(0,0,0,.07);"/>
-                    <input type="submit" class="button button-primary button-large"
-                           value="<?php echo esc_attr( __( 'Send', 'jetpack' ) ); ?>"/>
+                <p><label for="email">Your email<br/></label>
+                    <input type="email" name="email" class="text-input"/>
+                    <input type="submit" class="button"
+                           value="<?php echo esc_attr( __( 'Send email', 'jetpack' ) ); ?>"/>
```
Comment (jeherve) [inline modules/protect/blocked-login-page.php]: Should we use `esc_attr_e` here?

File: modules/protect/blocked-login-page.php [hunk 8]
```diff
@@ -258,72 +261,60 @@ function send_recovery_email() {
...
-            <p><?php _e( 'Email yourself a special link to regain access the login form.', 'jetpack' ); ?></p>
+        <div>
             <form method="post" action="?jetpack-protect-recovery=true">
 				<?php echo wp_nonce_field( 'bypass-protect' ); ?>
-                <p><label for="email" style="font-size:12px;">Email Address<br/></label>
-                    <input type="email" name="email" style="font-size:24px; padding:3px; margin: 2px 6px 16px 0; width:100%; border: 1px solid #ddd;
-    box-shadow: inset 0 1px 2px rgba(0,0,0,.07);"/>
-                    <input type="submit" class="button button-primary button-large"
-                           value="<?php echo esc_attr( __( 'Send', 'jetpack' ) ); ?>"/>
+                <p><label for="email">Your email<br/></label>
```
Comment (jeherve) [inline modules/protect/blocked-login-page.php]: I think we need to translate this as well.

**Comment by enejb** [comment]
Thanks for the review @jeherve

**Comment by rickybanister** [comment]
Yes, a notice like that @enejb, I'm not sure if dops-components has been updated to use those newish calypso styles though, so whatever is easiest to make it clear that there's been an error.

**Comment by enejb** [comment]
I am not relying on any external styles here. If you have some mock ups of how it should all look that would really help.

**Comment by rickybanister** [comment]
So this page has it's own css file?

This would work:

image

File: class.jetpack-constants.php [hunk 9]
```diff
@@ -31,6 +31,17 @@ public static function is_defined( $name ) {
...
 	}
+	/**
+	 * Checks if a "constant" has been set in Jetpack_Constants
+	 * and has the value of true
+	 *
+	 * @param $name string The name of the constant
+	 *
+	 * @return bool
+	 */
+	public static function is_true( $name ) {
```
Comment (lezama) [inline class.jetpack-constants.php]: ❤️

File: modules/protect/blocked-login-page.php [hunk 10]
```diff
@@ -220,9 +220,12 @@ public function render_blocked_login_message() {
...
 	function process_recovery_email() {
 		$sent = $this->send_recovery_email();
-
+		$show_recovery_form = true;
 		if ( is_wp_error( $sent ) ) {
-			$this->protect_die( $sent, true );
+			if ( 'email_already_sent' !== $sent->get_error_code() ) {
+				$show_recovery_form = true;
+			}
+			$this->protect_die( $sent,null,true, $show_recovery_form );
```
Comment (lezama) [inline modules/protect/blocked-login-page.php]: missing spaces after `,`s

**Comment by enejb** [comment]
cc: @oskosk I think this is ready once @rickybanister gives his blessing :)

**Comment by rickybanister** [comment]
There seems to be some extra unnecessary padding at the bottom of the form card:

image

image

The bottom padding should match the left and right padding of the card (24px).

My only other questions is how do we prevent this screen from receiving the same sort of DDOS attack as the user's login form:

image

Should we only allow them to try a few times?

**Comment by enejb** [comment]
> My only other questions is how do we prevent this screen from receiving the same sort of DDOS attack as the user's login form: 

We currently don't do anything here to solve that issue. I don't think that Jetpack Protect is also meant to solve that issue. It only prevents hackers from guessing passwords and not DDOS attacks.

If a user guesses the email address they will only be able to send the email to that address every 15 minutes. Since we don't allow the same email address to receive multiple requests if a valid email exists. 

cc: @samhotchkiss do you have any ideas or suggestions here?

**Comment by enejb** [comment]
Update screenshot with 24px padding.
screen shot 2017-12-08 at 11 35 53 am

**Comment by rickybanister** [comment]
🚢 it

**Comment by baeby** [comment]
Add

**Pull request merged**